monkey/common/event_queue/event_queue.py

```diff
@@ -0,0 +1,53 @@
+from typing import Any, Callable, Sequence
+
+from pubsub import pub
+
+from common.events import AbstractEvent
+
+from .i_event_queue import IEventQueue
+
+
+class EventQueue(IEventQueue):
+    @staticmethod
+    def subscribe_all(subscriber: Callable[..., Any]):
+        pub.subscribe(listener=subscriber, topicName=pub.ALL_TOPICS)
```

Comment:
Could this result in a subscriber receiving the event multiple times?

If an event has type `A` and tags `tag_1`, `tag_2`, then we'll publish the event 3 times. Once for topic "A", once for topic "tag_1", and once for topic "tag_2". I would expect that anything subscribed to `pub.ALL_TOPICS` would receive 3 identical events.

It's worth writing a unit test for this and verifying the behavior.

Reply:
Yes, the subscriber is triggered thrice.

Reply:
Nice use of "thrice" :)

Reply:
For now, the only time we'll be subscribing to all events is when sending the events to the Island. In that case, if we're sending the same event with different topic names, it makes sense to me, since different parts of the UI can be subscribed to different topics (type or tag(s)) of one event.

Reply:
We're only going to send each event to the Island once. The Island will have it's own EventQueue where components will be subscribed to what they need. The topic name does not get sent from the agent to the island.

Reply:
> We're only going to send each event to the Island once.

I'm not sure I completely understand. What comprises an "event" here? In other words, what exactly are we sending to the Island?

Regardless, we can have whatever is sending the events to the Island subscribe to all event types and no tags.

Reply:
> I'm not sure I completely understand. What comprises an "event" here? In other words, what exactly are we sending to the Island?

All events will subclass [AbstractEvent](https://github.com/guardicore/monkey/blob/b2fa790db8055050a4858078281c13b276314c08/monkey/common/events/abstract_event.py) and add their own attributes.

> Regardless, we can have whatever is sending the events to the Island subscribe to all event types and no tags.

The point of subscribing to all events is that you don't need to subscribe to types or tags. You subscribe to all events and you get all events. Publishing an event to a different topic (tag or type) does not constitute a new event.

Reply:
We could either check the topic name before sending an event to the Island (example of getting the topic name in a subscriber [here](https://github.com/guardicore/monkey/pull/2182#discussion_r941134053)), or we could modify the subscriber to check that it's not sending duplicate events.

Reply:
We can just create a topic name that's unlikely to get clobbered by tags and types and publish/subscribe to that.

```diff
-pub.subscribe(listener=subscriber, topicName=pub.ALL_TOPICS)
+pub.subscribe(listener=subscriber, topicName="INTERNAL_ALL_EVENTS_TOPIC")
```